sklearn/ensemble/_hist_gradient_boosting/binning.py

```diff
@@ -49,23 +50,47 @@ def _find_binning_thresholds(col_data, max_bins):
         col_data = col_data[~missing_mask]
     # The data will be sorted anyway in np.unique and again in percentile, so we do it
     # here. Sorting also returns a contiguous array.
-    col_data = np.sort(col_data)
+    sort_idx = np.argsort(col_data)
+    col_data = col_data[sort_idx]
     distinct_values = np.unique(col_data).astype(X_DTYPE)
-    if len(distinct_values) <= max_bins:
-        midpoints = distinct_values[:-1] + distinct_values[1:]
-        midpoints *= 0.5
+    if sample_weight is None:
+        if len(distinct_values) <= max_bins:
+            midpoints = distinct_values[:-1] + distinct_values[1:]
+            midpoints *= 0.5
+        else:
+            percentiles = np.linspace(0, 100, num=max_bins + 1)
+            percentiles = percentiles[1:-1]
+            midpoints = np.percentile(
```

Comment:
Same here, those are no longer midpoints since we use `method="average_inverted_cdf"`.